tests/conftest.py

```diff
@@ -643,45 +638,179 @@ def sqlite_cryptbank_connection() -> DatabaseContext:
     return DatabaseContext(DatabaseConnection(connection), DatabaseDialect.SQLITE)
 
 
+CUSTOM_DATASETS = ["synthea", "soccer_2016", "wdi", "keywords"]
+"""
+    Contains the name of all the custom datasets that will be used for testing.
+    This includes the datasets from S3 and initialized with a .sql file.
+"""
+CUSTOM_DATASETS_SCRIPTS = {
+    "keywords": "init_reserved_words_sqlite",
+    "wdi": "init_world_indicators_sqlite",
+}
+"""
+    Maps the datasets that need to be built with a sql script, with the name of
+    the script file. These datasets are NOT downloaded from S3.
+"""
+
+
+def get_s3_client() -> boto3.Session.client:
+    """
+    Generates an S3 client with the stablished credentials. For CI assumes OIDC
+    credentials provided by Github Actions
+
+    Returns:
+        The client of the created boto3 session
+    """
+    session: boto3.Session
+    if is_ci():
+        # Running in GitHub Actions CI — OIDC role will be assumed automatically
+        # Assuming in CI has aws-actions/configure-aws-credentials
+        session = boto3.Session()
+    else:
+        # Local development — use credentials stored in environment
+        # variables or ~/.aws/credentials
+        session = boto3.Session(
+            aws_access_key_id=os.getenv("AWS_ACCESS_KEY_ID"),
+            aws_secret_access_key=os.getenv("AWS_SECRET_ACCESS_KEY"),
+            region_name="us-east-2",
+        )
+
+    return session.client("s3")
+
+
+def get_s3_custom_datasets(
+    s3_client: boto3.Session.client,
+    data_folder: str,
+    metadata_folder: str,
+    datasets: list[str],
+    scripts: dict[str, str],
+) -> None:
+    """
+    Sets up the data and metadata for custom datasets testing. This includes
+    downloading the data (.db file) and metadata (.json file) from llm-fixtures
+    bucket and place them in data_folder and metadata_folder respectvely.Also,
+    includes executing the init script avaliable in CUSTOM_DATASETS_SCRIPTS,
+    when the script is executed the metadata must be created manually for testing
+    """
+
+    bucket: str = "llm-fixtures"
+
+    for dataset in datasets:
+        db_file: str = f"{data_folder}/{dataset}.db"
+
+        if dataset in scripts:
+            # setting up with script
+            # assuming the metadata is already in the repo
+            init_sql = f"{data_folder}/{scripts[dataset]}.sql"
+            subprocess.run(f"sqlite3 {db_file} < {init_sql}", shell=True, check=True)
+        else:
+            # download from s3
+            local_metadata_path: str = f"{metadata_folder}/{dataset}_graph.json"
+
+            key_data: str = f"data/{dataset}.db"
+            key_metadata: str = f"metadata/{dataset}.json"
```

Comment:
nit: move newline to be after key_metadata definition
```suggestion
           key_data: str = f"data/{dataset}.db"
            key_metadata: str = f"metadata/{dataset}.json"
            
```